src/libData/AccountData/AccountStore.cpp

```diff
@@ -179,19 +180,194 @@ void AccountStore::AddAccount(const PubKey& pubKey, const Account& account)
     AddAccount(Account::GetAddressFromPublicKey(pubKey), account);
 }
 
-void AccountStore::UpdateAccounts(const Transaction& transaction)
+void AccountStore::UpdateAccounts(const uint64_t& blockNum,
+                                  const Transaction& transaction)
 {
     //LOG_MARKER();
 
     const PubKey& senderPubKey = transaction.GetSenderPubKey();
     const Address fromAddr = Account::GetAddressFromPublicKey(senderPubKey);
-    const Address& toAddr = transaction.GetToAddr();
+    Address toAddr = transaction.GetToAddr();
     const uint256_t& amount = transaction.GetAmount();
 
+    if (transaction.GetCode().size() > 0 && toAddr == NullAddress)
+    {
+        // Create contract account
+        Account* account = GetAccount(fromAddr);
+        // TODO: remove this, temporary way to test transactions
+        if (account == nullptr)
+        {
+            AddAccount(fromAddr, {10000000000, 0});
+        }
+
+        toAddr = Account::GetAddressForContract(
+            fromAddr, m_addressToAccount[fromAddr].GetNonce());
+        AddAccount(toAddr, {0, 0});
+        m_addressToAccount[toAddr].SetCode(transaction.GetCode());
+        // Store the immutable states
+        m_addressToAccount[toAddr].InitContract(transaction.GetData());
+    }
+
+    if (transaction.GetData().size() > 0 && toAddr != NullAddress)
+    {
+        std::lock_guard<std::mutex> lk(m_mutexCallInterpreter);
+        // TODO: Trigger a contract account
+        // ParseJsonOutput(/*Call Interpreter To Get Json Output*/);
+        Account* fromAccount = GetAccount(fromAddr);
+        // TODO: remove this, temporary way to test transactions
+        if (fromAccount == nullptr)
+        {
+            AddAccount(fromAddr, {10000000000, 0});
+        }
+
+        Account* toAccount = GetAccount(toAddr);
+        if (toAccount == nullptr)
+        {
+            LOG_GENERAL(WARNING, "The target contract account doesn't exist");
+            return;
+        }
+
+        // TODO: Implement the calling of interpreter in multi-thread
+        if (ExportContractFiles(blockNum, toAccount, transaction))
+        {
+            SysCommand::ExecuteCmdWithOutput(GetContractCmdStr());
+            ParseContractOutput();
+        }
+    }
+
     TransferBalance(fromAddr, toAddr, amount);
     IncreaseNonce(fromAddr);
+}
+
+Json::Value AccountStore::GetBlockStateJson(const uint64_t& BlockNum) const
```

Comment:
I guess this function can easily be modified in case  blockstatejson has other values such as `TIMESTAMP` and `BLOCKHASH`?

Reply:
Yes, it's easy to modify